src/components/CompareResults/ResultsMain.tsx

```diff
@@ -91,18 +96,106 @@ function ResultsMain() {
     }
   }
 
+  /********** Edit Results Title Section **********/
+  const [rawSearchParams, updateRawSearchParams] = useRawSearchParams();
+  const initialComparisonTitle = rawSearchParams.get('title') ?? 'Results';
+  const [comparisonTitleName, setComparisonTitleName] = useState(
+    initialComparisonTitle,
+  );
+  const [previousComparisonTitle, setPreviousComparisonTitle] =
+    useState('Results');
```

Comment:
Usually if a state isn't used for the rendering, it's a good sign it shouldn't be a state. Indeed changing a state triggers a rerender, but that's not needed here. Therefore this should be a ref instead of a state.
Think of a `ref` just like a variable attached to this component instance.
More info about it: https://react.dev/reference/react/useRef